File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-cs/Models/OrderDetailDelivery.cs
namespace OctoSupply.Api.Models;

public sealed record OrderDetailDelivery(
    int OrderDetailDeliveryId,
    int OrderDetailId,
    int DeliveryId,
    int Quantity,
    string? Notes);

public sealed class CreateOrderDetailDeliveryRequest
{
    public int OrderDetailId { get; set; }
    public int DeliveryId { get; set; }
    public int Quantity { get; set; }
    public string? Notes { get; set; }
}

public sealed class UpdateOrderDetailDeliveryRequest
{
    public int? OrderDetailId { get; set; }
    public int? DeliveryId { get; set; }
    public int? Quantity { get; set; }
    public string? Notes { get; set; }
}

File: 03-Azure/01-04-AI/01_Agentic_SDLC/src/api-py/models/order_detail_delivery.py
from __future__ import annotations

from pydantic import BaseModel


class OrderDetailDelivery(BaseModel):
    orderDetailDeliveryId: int
    orderDetailId: int
    deliveryId: int
    quantity: int
    notes: str | None = None


class CreateOrderDetailDeliveryRequest(BaseModel):
    orderDetailId: int
    deliveryId: int
    quantity: int
    notes: str | None = None


class UpdateOrderDetailDeliveryRequest(BaseModel):
    orderDetailId: int | None = None
    deliveryId: int | None = None
    quantity: int | None = None
    notes: str | None = None
